rxsmartlock/src/main/java/com/freeletics/rxsmartlock/SmartLockComponent.kt

```diff
@@ -279,4 +284,12 @@ internal class SmartLockComponent {
         activityResultSubject.onNext(ActivityResult(requestCode, resultCode, resultData))
         isDialogShown = false
     }
+
+    /**
+     * Smart lock operation error.
+     */
+    class SmartLockException(
+        message: String,
```

Comment:
What do you think about introducing error codes here? Then it could be easier handled from the app.

Reply:
Could be either this or more different exception classes. Both sounds fine for me.

Reply:
Yes, different exception class is also fine 👍

Reply:
@dkhmelenko I've introduced `reason` field in `SmartLockException` custom exception class